Add incomplete_package_import option to speed reposync up

moio opened the pull request:
Considerable time is taken by `spacewalk-repo-sync` and inter-server sync due to database insertions in tables with a high number of rows.

Three of these (`rhnPackageFile`, `rhnPackageChangeLogRec` and `rhnPackageChangeLogData`) only contain informational data that can be skipped in order to get faster repo-sync times, at the expense of not finding this data in the Web UI.

This patch adds an `rhn.conf` flag (`incomplete_package_import = 1`) that will result in `spacewalk-repo-sync` and inter-server sync skipping those tables with noticeable performance gains.

This is particularly useful in tests and demos, and is not recommended in production.

eherget commented:
Please update the rhn.conf man page with this new config item.  Man page is here in the source tree spacewalk/spacewalk/config/usr/share/man/man5/rhn.conf.5

grinrag commented:
Hi Silvio,
What is the name of parameter  'fast_package_import = 1' or 'incomplete_package_import = 1'?
Please, add into commit message BZ number 1447296.

moio commented:
@grinrag PR comment updated to reflect code
@eherget man page section added

I also added a subsequent commit that addressed a problem we discovered later.

grinrag commented:
Silvio,
1) Is there a big reason to skip package changelog importing? How much time can we save, have you made some tests?

2) Maybe, in this case, rename config parameter into 'skip_package_changelog'?

3) Squash all commits into the first one and commit message should be "1447296 - ... "

moio commented:
@grinrag in my tests I get a 10-25% speedup depending on hardware (eg. rotating disks vs SSDs) and repos.

I also squashed the commits, added the bug reference and renamed the option.

Hope it's good to merge now!

grinrag commented:
Thank you. Merged.
